src/mame/machine/tsconf.cpp

```diff
@@ -72,218 +75,406 @@ void tsconf_state::tsconf_update_bank1()
 		m_ROMSelection |= (m_regs[PAGE0] & 0xfc);
 	}
 
-	uint8_t *rom0;
 	if (W0_RAM)
 	{
-		rom0 = m_ram->pointer() + PAGE4K(m_ROMSelection);
-		m_bank1->set_base(rom0);
+		m_banks[0]->configure_entries(0, m_ram->size() / 0x4000, m_ram->pointer(), 0x4000);
+		m_banks[0]->set_entry(m_ROMSelection);
 	}
 	else
 	{
-		rom0 = &m_p_rom[0x10000 + PAGE4K(m_ROMSelection & 0x1f)];
-		m_bank1->set_base(rom0);
+		memory_region *rom = memregion("maincpu");
+		u8 rom_banks = (rom->bytes() - 0x10000) / 0x4000;
+		m_banks[0]->configure_entries(0, rom_banks, rom->base() + 0x10000, 0x4000);
+		m_banks[0]->set_entry(m_ROMSelection & 0x1f);
```

Comment:
Rather than reconfiguring `m_banks[0]`, could you use a `memory_view` for enabling and disabling the ROM overlay?  That way you could have a separate `memory_bank` for the ROM banks, all the configuration could be done at machine start, and the ROM/RAM selection would be saved automatically.

Reply:
At the first look that's not going to work.
bank0 can be configured as a page selector either from ROM or RAM.
I'll try later unless you agree with description above.

Reply:
Thank you for the education!